### PR TITLE
Change summaries to histograms

### PR DESCRIPTION
**Pull request by grobie**
@fabxc @beorn7

### CODE REVIEW

File: eagle.go [hunk 1]
```diff
@@ -69,17 +69,18 @@ func main() {
...
-			prometheus.SummaryOpts{
+		latencies = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
 				Namespace:   namespace,
-				Name:        "request_durations_nanoseconds",
-				Help:        "The total duration of HTTP requests (nanoseconds).",
+				Name:        "request_duration_seconds",
+				Help:        "The total duration of HTTP requests (seconds).",
 				ConstLabels: constLabels,
+				Buckets:     []float64{.00001, .0001, .001, .01, .1, 1, 10},
```
Comment (beorn7) [inline eagle.go]: Perhaps use the ExponentialBuckets convenience function?

File: squirrel/squirrel.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func main() {
 
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		defer func(began time.Time, r *http.Request) {
-			duration := float64(time.Since(began))
+			duration := float64(time.Since(began)) / float64(time.Second)
```
Comment (beorn7) [inline squirrel/squirrel.go]: Just `time.Since(began).Seconds()`.

File: eagle.go [hunk 3]
```diff
@@ -96,7 +97,7 @@ func (r *registry) collect(resultc chan result) {
 		for _, t := range rr.results {
 			r.latencies.
 				WithLabelValues(rr.target, strconv.Itoa(int(t.Code)), rr.endpoint).
-				Observe(float64(t.Latency.Nanoseconds()))
+				Observe(float64(t.Latency.Nanoseconds()) / float64(time.Second))
```
Comment (beorn7) [inline eagle.go]: If t.Latency is a time.Duration, use the Seconds() method.

**Comment by grobie** [comment]
@beorn7 updated all of them. here the result of a local test:

```
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="1e-05"} 0
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="0.0001"} 0
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="0.001"} 16356
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="0.01"} 16699
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="0.1"} 16700
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="1"} 16700
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="10"} 16700
eagle_request_duration_seconds_bucket{code="200",endpoint="http://0:7801",target="direct",test="unknown",le="+Inf"} 16700
```

File: eagle.go [hunk 4]
```diff
@@ -96,7 +97,7 @@ func (r *registry) collect(resultc chan result) {
 		for _, t := range rr.results {
 			r.latencies.
 				WithLabelValues(rr.target, strconv.Itoa(int(t.Code)), rr.endpoint).
-				Observe(float64(t.Latency.Nanoseconds()))
+				Observe(float64(t.Latency.Seconds()))
```
Comment (beorn7) [inline eagle.go]: Seconds() already returns a float64.

File: squirrel/squirrel.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func main() {
 
 	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
 		defer func(began time.Time, r *http.Request) {
-			duration := float64(time.Since(began))
+			duration := float64(time.Since(began).Seconds())
```
Comment (beorn7) [inline squirrel/squirrel.go]: Seconds() already returns a float64.

File: eagle.go [hunk 6]
```diff
@@ -69,17 +69,18 @@ func main() {
...
-			prometheus.SummaryOpts{
+		latencies = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
 				Namespace:   namespace,
-				Name:        "request_durations_nanoseconds",
-				Help:        "The total duration of HTTP requests (nanoseconds).",
+				Name:        "request_duration_seconds",
+				Help:        "The total duration of HTTP requests (seconds).",
 				ConstLabels: constLabels,
+				Buckets:     prometheus.ExponentialBuckets(.00001, 10.0, 7),
```
Comment (beorn7) [inline eagle.go]: As discussed elsewhere, hand-crafted buckets might be better (with highest resolution around the interesting range of ~2ms). Or start at ~0.5ms and use a factor of 1.5 or 2.0.

**Comment by grobie** [comment]
@beorn7 Updated all of them. I kicked out the squirrel histogram/summary completely as it's actually not interesting to know how long it takes to respond with a 200. I changed the histogram for the eagle measured latency to use only a few buckets around the interesting values.

**Comment by fabxc** [comment]
:+1:

**Comment by beorn7** [comment]
:+1:

**Pull request merged**